applications/nrf_desktop/src/events/hid_event.c

```diff
@@ -8,36 +8,44 @@
 
 #include "hid_event.h"
 
+#define HID_EVENT_LOG_BUF_LEN 128
+
 
 static int log_hid_report_event(const struct event_header *eh, char *buf,
 				size_t buf_len)
 {
 	const struct hid_report_event *event = cast_hid_report_event(eh);
 	int pos;
+	char log_buf[HID_EVENT_LOG_BUF_LEN];
 
 	__ASSERT_NO_MSG(event->dyndata.size > 0);
 
-	pos = snprintf(buf, buf_len, "Report 0x%x src:%p sub:%p:",
+	pos = snprintf(log_buf, sizeof(log_buf), "Report 0x%x src:%p sub:%p:",
 		       event->dyndata.data[0],
 		       event->source,
 		       event->subscriber);
-	if ((pos > 0) && (pos < buf_len)) {
+	if ((pos > 0) && (pos < sizeof(log_buf))) {
 		for (size_t i = 1; i < event->dyndata.size; i++) {
-			int tmp = snprintf(&buf[pos], buf_len - pos,
+			int tmp = snprintf(&log_buf[pos], sizeof(log_buf) - pos,
 					   " 0x%.2x", event->dyndata.data[i]);
 			if (tmp < 0) {
+				log_buf[sizeof(log_buf) - 2] = '~';
 				pos = tmp;
 				break;
 			}
 
 			pos += tmp;
 
-			if (pos >= buf_len)
+			if (pos >= sizeof(log_buf))
 				break;
 		}
 	}
-
-	return pos;
+	if (pos < 0) {
+		EVENT_MANAGER_LOG(eh, "hid_report_event Event Manager logging function failed");
```

Comment:
let's use the same simple message as in the other file 
the name of the event will be added by macro